apps/site-nft/src/components/AuthRoute/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+import { useActiveAccount } from '@thecointech/shared/containers/AccountMap';
+import { Login } from "@thecointech/shared/containers/Login";
+import { isWallet } from "@thecointech/shared/SignerIdent";
+import { FormattedMessage } from 'react-intl';
+import { useAccount } from '@thecointech/shared/containers/Account';
+
+const waitingForWeb3 = {
+  defaultMessage: "Connecting to your Web3 provider",
+  description:"Message to display while waiting for user to complete Web3 connection"
+};
+
+// An authorized route is one that requires the use of an unlocked account
+export const AuthRoute = (routeProps: RouteProps)  => {
```

Comment:
This is the other thing that is significantly different from app.  I originally built the `Account.tsx` class to handle all the account management.  We passed it in a map of all the different children.

Unfortunately, it didn't place nicely with this app, because it had a lot of opinions (ie - it needed a sidebar).  It was doing a lot of jobs at once, and every time I went back to it it felt very confusing.

This component does nearly the same thing, but as an `AuthRoute` attached to a switcher.  It just makes sure that the account is valid before showing the component.  It's a lot nicer to reason about, because it doesn't do tonnes of weird stuff like setup menus.